applications/plugins/SofaCUDA/sofa/gpu/cuda/CudaBarycentricMapping.h

```diff
@@ -131,7 +131,7 @@ class BarycentricMapperSparseGridTopology< gpu::cuda::CudaVectorTypes<VecIn,VecI
     Index addPointInCube(const Index cubeIndex, const SReal* baryCoords);
 
     bool isEmpty() { return map.size() == 0; }
-    void setTopology(topology::RegularGridTopology* _topology) { topology = _topology; }
+    void setTopology(topology::SparseGridTopology* _topology) { topology = _topology; }
```

Comment:
https://github.com/sofa-framework/sofa/blob/aaf750f3ad2dce0aee6e42ff4e896801015afa11/applications/plugins/SofaCUDA/sofa/gpu/cuda/CudaBarycentricMapping.h#L58

Not sure to understand your change. topology is a RegularGridTopology* in this class.

Reply:
I think it is more this one (which is not a RegulardGridTopology):
https://github.com/sofa-framework/sofa/blob/aaf750f3ad2dce0aee6e42ff4e896801015afa11/applications/plugins/SofaCUDA/sofa/gpu/cuda/CudaBarycentricMapping.h#L109

Reply:
if you look at the name of the class BarycentricMapperSparseGridTopology, it would make sense to set a ..... SparseGridTopology anyway. To me, it looks like clearly a copy-paste too fast